src/Microsoft.AspNet.Mvc.Razor/VirtualPathRazorPageFactory.cs

```diff
@@ -60,10 +62,12 @@ public IRazorPage CreateInstance([NotNull] string relativePath)
                 return null;
             }
 
-            var page = (IRazorPage)_activator.CreateInstance(_serviceProvider, result.CompilationResult.CompiledType);
+            var razorPageFactory = _razorPageCache.GetOrAdd(result.CompilationResult.CompiledType,
+                _createFactory);
+            var page = (IRazorPage)razorPageFactory(_serviceProvider, null);
```

Comment:
This would be weird - you'd be activating a view with application services, not request. Either 

1. We need to flow `ViewContext` and use that to get `RequestServices`
2. We should assume Views don't need to be type activated and should use `Activator.CreateInstance`